src/main/java/com/example/demo/dao/GradeLocalRepository.java

```diff
@@ -0,0 +1,73 @@
+package com.example.demo.dao;
+
+import com.example.demo.dao.entity.Grade;
+import org.springframework.stereotype.Repository;
+
+import java.util.*;
+
+@Repository
+public class GradeLocalRepository implements GradeRepository {
+    private Map<Long, List<Grade>> gradeRegister;
+
+    public GradeLocalRepository() {
+        this.gradeRegister = new HashMap<>();
+    }
+
+    @Override
+    public List<Grade> getAllGrades() {
```

Comment:
Tutaj pewnie da sie to uprosic forEachem - kazdą liste wkladac cała przez addAll do wynikowej listy
Zastanaiwa mnie bardziej czy jest taka potrzeba takiej listy, a jesli jest czy nie wygodniej by bylo aby kazdy student mial swoja listę ocen. W twojej wersji kazda ocena przypisana jest do studenta, nie wiem czy jest potrzeba zeby każda ocena wiedziala do jakiego studenta została przypisana. Spróbuj zrobić ocene jako enum, nie ma znaczenia 5tka Roberta, a 5tka Sebastiana, w obu przypadkach to ta sama 5tka

Reply:
Funkcja wygląda inaczej w aktualnej wersji aplikacji, problem rozwiązany jest z wykorzystaniem stream API. Obiekty Grade zawierają teraz więcej informacji takich jak waga oceny czy data dodania, co nadaje sens im istnieniu jako osobna klasa.